integration_tests/binding/binding.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package binding contains an Ondatra binding wrapping knebind for lemming.
+package binding
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"text/template"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/openconfig/ondatra/binding"
+	"github.com/openconfig/ondatra/proto"
+	"google.golang.org/protobuf/encoding/prototext"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+
+	tpb "github.com/openconfig/kne/proto/topo"
+	kinit "github.com/openconfig/ondatra/knebind/init"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const config = `
+topology: {{  .Topology }}
+kubecfg: {{ .Kubeconfig }}
+cli: {{ .CLI }}
+username: {{ .Username }}
+password: {{ .Password }}
+`
+
+type configFields struct {
+	Topology   string
+	Kubeconfig string
+	CLI        string
+	Username   string
+	Password   string
+}
+
+var (
+	keep = flag.Bool("keep", false, "Keep topology deployed after test")
+)
+
+// Get returns the custom lemming binding.
+func Get(topoDir string) func() (binding.Binding, error) {
+	dir, _ := filepath.Abs(topoDir)
+	testbedFile := filepath.Join(dir, "testbed.pb.txt")
+	topoFile := filepath.Join(dir, "topology.pb.txt")
+
+	flag.Set("testbed", testbedFile)
+
+	return func() (binding.Binding, error) {
+		fmt.Println("Checking for KNE CLI installation")
+		knePath, err := getKNECLIPath()
+		if err != nil {
+			return nil, err
+		}
+
+		fmt.Println("Creating knebind config file")
+		if err := writeKNEBindCfg(knePath, topoFile); err != nil {
+			return nil, err
+		}
+		b, err := kinit.Init()
+		if err != nil {
+			return nil, err
+		}
+		return &LemmingBind{
+			Binding:  b,
+			topoFile: topoFile,
+			kubecfg:  os.ExpandEnv("$HOME/.kube/config"),
+		}, nil
+	}
+}
+
+// LemmingBind wraps the Ondatra knebind and adds the ability to set the testbed and topology from inside the test.
+// TODO: Add unit tests and upstream some of this to Ondatra.
```

Comment:
Are you planning to export all of this to knebind? So perhaps pick up the testbed/topology, and create the config YAML **if** the user doesn't pass in these are arguments? This seems like a useful thing whenever running KNE tests.

Reply:
I was planning on prototyping the changes here first (where we don't have to worry about backwards-compat) then discuss to see if there's some parts of this we want to upstream